File: src/main/java/cz/stasimek/fakturaceeasypeasy/repository/UserRepository.java
package cz.stasimek.fakturaceeasypeasy.repository;

import cz.stasimek.fakturaceeasypeasy.entity.User;
import cz.stasimek.fakturaceeasypeasy.enumeration.Provider;
import java.util.UUID;
import org.springframework.data.repository.CrudRepository;

public interface UserRepository extends CrudRepository<User, UUID> {

	User findByProviderAndLogin(Provider provider, String login);

	User findByProviderAndEmail(Provider provider, String email);
}

File: src/main/kotlin/cz/stasimek/fakturaceeasypeasy/repository/UserRepository.kt
package cz.stasimek.fakturaceeasypeasy.repository

import cz.stasimek.fakturaceeasypeasy.entity.User
import cz.stasimek.fakturaceeasypeasy.enumeration.Provider
import org.springframework.data.repository.CrudRepository
import java.util.*

interface UserRepository : CrudRepository<User, UUID> {

    fun findByProviderAndLogin(provider: Provider, login: String): User?

    fun findByProviderAndEmail(provider: Provider, email: String): User?
}